package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces

```diff
@@ -90,8 +90,15 @@ else
 	sysconfig.single_ifname = lan_ifname or wan_ifname
 end
 
+-- Delete all UCI device sections of type 'bridge'
+-- as well as the ones starting with 'br-'.
+-- Preserve all others to apply MAC address stored in UCI
+uci:foreach('network', 'device',function(s)
+	if s.type == 'bridge' or s.name:match('^br-') then
+		uci:delete('network', s['.name'])
+	end
+end)
```

Comment:
Using `delete` in `foreach` has caused weird breakage in the past (including segfaulting Lua at some point). `delete_all` with a function argument must be used instead, see `gluon-respondd/luasrc/lib/gluon/upgrade/400-respondd-firewall`

Reply:
I'll check that